instrumentation/mysql2/lib/opentelemetry/instrumentation/mysql2/patches/client.rb

```diff
@@ -95,20 +95,23 @@ def detect_unmatched_pairs(obfuscated)
             %r{'|"|\/\*|\*\/}.match(obfuscated)
           end
 
-          def database_span_name(sql)
-            # Setting span name to the SQL query without obfuscation would
-            # result in PII + cardinality issues.
-            # First attempt to infer the statement type then fallback to
-            # current Otel approach {database.component_name}.{database_instance_name}
-            # https://github.com/open-telemetry/opentelemetry-python/blob/39fa078312e6f41c403aa8cad1868264011f7546/ext/opentelemetry-ext-dbapi/tests/test_dbapi_integration.py#L53
-            # This creates span names like mysql.default, mysql.replica, postgresql.staging etc.
-
-            statement_type = extract_statement_type(sql)
-
-            return statement_type unless statement_type.nil?
-
-            # fallback
-            database_name ? "mysql.#{database_name}" : 'mysql'
+          def database_span_name(sql) # rubocop:disable Metrics/CyclomaticComplexity
+            case config[:span_name]
+            when :statement_type
+              extract_statement_type(sql)
+            when :db_name
+              database_name
```

Comment:
Not a new problem and hence not a blocker, but worth flagging that `database_name` could be an empty string (see code [here](https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/222/files#diff-09613a195baead8b773b7041065d5878ed88c7742a035db6e9bee436cbc05c0eR119)).

Reply:
Empty string or `nil`? The `nil` case is handled below with `|| 'mysql'`. I'm not sure which code you linked to.

Reply:
I was reading the code wrong, but was looking at the [database_name definition](https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/main/instrumentation/mysql2/lib/opentelemetry/instrumentation/mysql2/patches/client.rb#L117-L120C14).  It can result in a `nil` (`(nil)&.to_s => nil`), which is handled, here.